bgpd/bgp_community.c

```diff
@@ -442,17 +429,15 @@ static void set_community_string(struct community *com, bool make_json)
 			}
 			break;
 		case COMMUNITY_LOCAL_AS:
-			strcpy(pnt, "local-AS");
-			pnt += strlen("local-AS");
+			strlcat(pnt, "local-AS", len);
 			if (make_json) {
 				json_string = json_object_new_string("localAs");
 				json_object_array_add(json_community_list,
 						      json_string);
 			}
 			break;
 		case COMMUNITY_NO_PEER:
-			strcpy(pnt, "no-peer");
-			pnt += strlen("no-peer");
+			strlcat(pnt, "no-peer", len);
 			if (make_json) {
 				json_string = json_object_new_string("noPeer");
 				json_object_array_add(json_community_list,
```

Comment:
Beyond this point, looks like default case needs some change. There is a sprintf on the pnt. pnt in the older version is for tracking the end of so far constructed. now that pnt doesn't track it, it could be an issue.

Reply:
You're correct, good catch